### DISCUSSION TITLE
Encoder like a analog note

### DISCUSSION THREAD

**Discussion by pableitordasauto** [shanteacontrols/OpenDeck#269] [Ideas]
Hi paradajz,
I see that in the fader configuración, you can send a note on with floating value. 
In encoders, you can send an incremental cc.
A good feature will be send the incremental value for the encoders like a floating note value between 0.00 and 1.00.
What do you think? I suppose that only needs a change in the midi packet header value.
Thanks!!

**Comment by pmjcontroller1** [comment]
That would be very good, but the ideal would be that the encoders could be configured as 2 different notes as if they were a button for each direction of rotation and with each pulse that would repeat the note.

**Comment by paradajz** [comment]
I'm not sure what exactly are you talking about. There are no float values in MIDI protocol.

**Reply by pableitordasauto** [reply]
Sorry for my poor explanation. 
I would mean:
Recognize the cw and ccw in encoders.
Then two options to send midi.
Option 1:
Instead CC, send midi note value like analog button
Option 2:
If the encoder turns cw, send note on-note off in each encoder event
If the encoder turns ccw, send other note on-note off in each encoder event
There are some software that works encoders with notes instead CC packets

**Comment by paradajz** [comment]
Option 1 is now available on master branch. Rotations send the same note with increasing or decreasing velocity, depending on the direction.

**Reply by pmjcontroller1** [reply]
Hello Paradax, I have tried this change that you have implemented, it would be possible to add the option to differentiate the direction of rotation with a different note. 

 Example 
ccw note 41 
cc note 42

**Reply by paradajz** [reply]
Maybe in the future.

**Reply by pmjcontroller1** [reply]
Hello Parax, this implementation would be of great help for lighting software that is limited to on/off note, so we could use encoders in all these software, do you think it will be in the near future

**Reply by paradajz** [reply]
Are you willing to contribute financially to the project?

Links of note:
https://github.com/shanteacontrols/OpenDeck/discussions/240#discussioncomment-7338210
https://github.com/shanteacontrols/OpenDeck/discussions/228#discussioncomment-6849361

Two mentions of donations.

Especially:

> "(...) we manufacture your L plates. Once they work, we plan to make you a donation, since we would include your project in our systems

Written 7 months ago.

I see that you already sell a product based on my work:
https://pmjcontroller.es/producto/circus-wing-midi-para-ma3/

I am not saying that you cannot do that (you can) but please don't request new features so that you can sell more of your product while promising donations which never happened. This is disingenuous.

**Reply by pmjcontroller1** [reply]
Hi paradajz, 

We have answered by email this morning to discuss the donations.
Please, confirm that you received it.
We tried to manage this collaboration before but we didn't get any reply.
Adding, the web doesn't have any link to donate.
We are always opened to reward your work. 
Only we need a meeting, mailing or something else that you want to make the better way to submit the collaboration.
Thanks!!!
~~~~~~~~

**Reply by paradajz** [reply]
> We have answered by email this morning to discuss the donations.

I haven't received anything.

**Reply by pmjcontroller1** [reply]
Ok this is the problem. ![Screenshot_2024-03-31-12-10-27-27_93b7fc0429a0e97afffb8ed9ac823d5b.jpg](https://github.com/shanteacontrols/OpenDeck/assets/118101014/564dd90b-ca37-4506-9c3b-f230e3999e7e)

![Screenshot_2024-03-31-12-10-35-71_93b7fc0429a0e97afffb8ed9ac823d5b.jpg](https://github.com/shanteacontrols/OpenDeck/assets/118101014/c378500a-0adb-4e45-a554-3baecb3288d0)

These are the emails we sent you yesterday.  That was the problem, we responded directly to the email on two previous occasions to ask you for a meeting because we couldn't make donations on your channel.
Please contact us, and we'll talk about it.

**Reply by pmjcontroller1** [reply]
![Screenshot_2024-03-31-12-15-03-26_99c04817c0de5652397fc8b56c3b3817.jpg](https://github.com/shanteacontrols/OpenDeck/assets/118101014/d6d2383d-d334-461e-af59-6705f3b50978)

![Screenshot_2024-03-31-12-14-50-73_99c04817c0de5652397fc8b56c3b3817.jpg](https://github.com/shanteacontrols/OpenDeck/assets/118101014/da714633-9904-452e-a5e1-7c0cc8778f2b)

The sponsor heart does not appear in your github, while in the others we can.

**Comment by paradajz** [comment]
The encoders now support configuration of two notes and the value for that note. There's also single-note repeated mode available which will send the same note+value in each direction, and another mode where one direction sends configured note+value and another direction sends the same note but with value 0.

**Comment by pableitordasauto** [comment]
Thanks igor.
I'm going to check it.

**Comment by pableitordasauto** [comment]
Hi Igor,

I have doubts.
The updated firmware is version7.9.0? This is the last firmware in the
releases page. Because I don't be sure...I have updated the board to test
anyway.

In the web configurator...I have tested the "Two note" but it works like
"Single Note/ variable value"
Besides, how can I configure the note for cw and note for ccw?

[image: image.png]

And I have a question. Do you have a board to test? Maybe we can send you
one in case that you don't have any.

**Comment by paradajz** [comment]
The firmware is not released yet. The code is on the master branch so you'd need to compile it.

**Comment by pableitordasauto** [comment]
Hi,

Then, I must execute the docker and follow the procedure, isn't it?
Take in mind that I don't be an expert user in Linux...

**Comment by paradajz** [comment]
[opendeck31.zip](https://github.com/shanteacontrols/OpenDeck/files/15015301/opendeck31.zip)

I've attached both sysex and uf2 firmware variants for the 3.1 board.

**Comment by pableitordasauto** [comment]
Hi,
Finally I can made the firmware. It works fine!!!
Many thanks!!!
I'm going to test more to be ensure about the functionality.
We'll be in contact. We have more ideas to develop the firmware.
It could be a win win future!!!

**Comment by paradajz** [comment]
Glad it works out for you. Released as a part of the new firmare, v7.10.0:
https://github.com/shanteacontrols/OpenDeck/releases/tag/v7.10.0
